Add 'Cleanup Snapshot' API Documentation (#8373)
opensearch-project/documentation-website: _api-reference/snapshots/cleanup-snapshot-repository.md

Introduced 1.0
{: .label .label-purple }

The Cleanup Snapshot Repository API clears a snapshot repository of data no longer referenced by any existing snapshot.

## Path and HTTP methods

```json
POST /_snapshot/<repository>/_cleanup
```
{% include copy.html %}

## Path parameters

| Parameter | Data type | Description |
| :--- | :--- | :--- |
| `repository` | String | The name of the snapshot repository. |

## Query parameters

The following table lists the available query parameters. All query parameters are optional.

| Parameter |  Data type | Description |
| :--- | :--- | :--- |
| `cluster_manager_timeout` | Time | The amount of time to wait for a response from the cluster manager node. Formerly called `master_timeout`. Optional. Default is 30 seconds. |
| `timeout` | Time | The amount of time to wait for the operation to complete. Optional.|

## Example request

The following request removes all stale data from the repository `my_backup`:

```json
POST /_snapshot/my_backup/_cleanup
```
{% include copy-curl.html %}

## Example response

```json
{
	"results":{
		"deleted_bytes":40,
		"deleted_blobs":8
	}
}
```

## Response body fields

| Field | Data type | Description |

> | :--- | :--- | :--- |

Reviewer: Below: Can we use something more precise than "freed"? Do we mean something like "The number of bytes made available by data deletion"?

> | `deleted_bytes` | Integer | The amount of bytes freed after data deletion. |

Reviewer: ```suggestion
| `deleted_bytes` | Integer | The number of bytes freed after data deletion. |
```

> | `deleted_bytes` | Integer | The amount of bytes freed after data deletion. |

Reviewer 2: @natebower: Does this work?

```suggestion
| `deleted_bytes` | Integer | The number of bytes now available in the snapshot after data deletion. |
```

> | `deleted_bytes` | Integer | The number of bytes now available in the snapshot after data deletion. |

Reviewer 2: ```suggestion
| `deleted_bytes` | Integer | The number of bytes made available in the snapshot after data deletion. |
```

> | `deleted_blobs` | Integer | The amount of binary large objects (BLOBs) cleared from the repository by the request. |

Reviewer: ```suggestion
| `deleted_blobs` | Integer | The number of binary large objects (BLOBs) cleared from the repository by the request. |
```
